cmd/up/up.go

```diff
@@ -906,6 +887,35 @@ func (up *upContext) runCommand(ctx context.Context) error {
 	)
 }
 
+func (up *upContext) checkOktetoStartError(ctx context.Context, msg string) error {
+	userID := pods.GetDevPodUserID(ctx, up.Dev, up.Client)
+	if up.Dev.PersistentVolumeEnabled() {
+		if userID != -1 && userID != *up.Dev.SecurityContext.RunAsUser {
+			return errors.UserError{
+				E:    fmt.Errorf("User %d doesn't have write permissions for the %s directory", userID, up.Dev.MountPath),
+				Hint: fmt.Sprintf("Set 'securityContext.runAsUser: %d' in your okteto manifest\n    After that, run 'okteto down -v' to reset your development container and run 'okteto up' again", userID),
+			}
+		}
+	} else {
+		if pods.OktetoDevPodMustBeRecreated(ctx, up.Dev, up.Client) {
```

Comment:
it's odd that the function called `checkOktetoStartError` changes the system. can we split the destruction into a separate stage? (or return a speficic error, and then the mail loop does the recreation?

Reply:
ok